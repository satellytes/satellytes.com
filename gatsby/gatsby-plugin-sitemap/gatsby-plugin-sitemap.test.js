const { resolvePages, serialize } = require('./resolve-pages');

const EXAMPLES_NODES = {
  allSitePage: {
    nodes: [
      { path: '/some-page' },
      { path: '/de/some-page' },
      { path: '/de/only-german' },
      { path: '/only-default' },
    ],
  },
};
describe('Gatsby Plugin Sitemap', () => {
  test('serialize creates a correct sitemap entry', () => {
    const result = serialize({
      path: '/some-page',
      language: 'en',
      links: [
        {
          lang: 'en',
          path: 'abc',
        },
        {
          lang: 'de',
          path: 'de/abc',
        },
      ],
    });

    expect(result).toEqual({
      url: '/some-page',
      changefreq: 'daily',
      priority: 0.7,
      links: [
        {
          lang: 'en',
          url: 'abc',
        },
        {
          lang: 'de',
          url: 'de/abc',
        },
      ],
    });
  });

  test('resolve transforms page nodes into sitemap items', () => {
    const result = resolvePages(EXAMPLES_NODES);

    expect(result).toEqual([
      { path: '/some-page', languageKey: 'en', normalizedPath: '/some-page' },
      {
        path: '/de/some-page',
        languageKey: 'de',
        normalizedPath: '/some-page',
      },
      {
        path: '/de/only-german',
        languageKey: 'de',
        normalizedPath: '/only-german',
      },
      {
        path: '/only-default',
        languageKey: 'en',
        normalizedPath: '/only-default',
      },
    ]);
    // test("resolve transforms page nodes into sitemap items", () => {
    //   const result = resolvePages(EXAMPLES_NODES)
    //   expect(result).toEqual([
    //     { path: '/some-page', translation: '/de/some-page' },
    //     { path: '/de/some-page', translation: '/some-page' },
    //     { path: '/de/only-german', translation: undefined },
    //     { path: '/only-default', translation: undefined }
    //   ]);
  });
});
