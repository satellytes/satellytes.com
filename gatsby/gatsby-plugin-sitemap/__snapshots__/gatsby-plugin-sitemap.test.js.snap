// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gatsby Plugin Sitemap works 1`] = `
Array [
  Object {
    "links": Array [
      Object {
        "lang": "en",
        "url": "/some-page/",
      },
      Object {
        "lang": "de",
        "url": "/de/some-page/",
      },
    ],
    "url": "/some-page/",
  },
  Object {
    "links": Array [
      Object {
        "lang": "en",
        "url": "/some-page/",
      },
      Object {
        "lang": "de",
        "url": "/de/some-page/",
      },
    ],
    "url": "/de/some-page/",
  },
  Object {
    "links": Array [
      Object {
        "lang": "de",
        "url": "/de/only-german/",
      },
    ],
    "url": "/de/only-german/",
  },
  Object {
    "links": Array [
      Object {
        "lang": "en",
        "url": "/only-default/",
      },
    ],
    "url": "/only-default/",
  },
]
`;
